script/copy_deduplicate

```diff
@@ -0,0 +1,145 @@
+)
+parser.add_argument(
+    "--dry-run",
+    action="store_true",
+    help=(
+        "Do not run queries, but instead print the query job config "
+        "and bytes that would be processed"
+    ),
+)
+group = parser.add_mutually_exclusive_group()
+group.add_argument(
+    "--only",
+    nargs="+",
+    dest="only_tables",
+    help=(
+        "Process only the given tables; "
+        "pass names or globs like 'telemetry_live.main_v*' "
+    ),
+)
+group.add_argument(
+    "--except",
+    nargs="+",
+    dest="except_tables",
+    help=(
+        "Process all tables in *_live datasets except for the given tables; "
+        "pass names or globs like 'telemetry_live.main_v*'"
+    ),
+)
+
+
+def sql_full_table_id(table):
+    return table.full_table_id.replace(":", ".")
+
+
+def run_deduplication_query(client, live_table, stable_table, date, dry_run):
+
+    sql = QUERY_TEMPLATE.format(source_table_spec=sql_full_table_id(live_table))
+    destination = f"{sql_full_table_id(stable_table)}${date:%Y%m%d}"
+    job_config = bigquery.QueryJobConfig(
+        destination=destination,
+        query_parameters=[
+            bigquery.ScalarQueryParameter("submission_date", "DATE", date)
+        ],
+        use_legacy_sql=False,
+        write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,
+        dry_run=dry_run,
+    )
+    query_job = client.query(sql, job_config)
+    if dry_run:
+        print(
+            "Would process {} bytes: {}".format(
+                query_job.total_bytes_processed, query_job.to_api_repr()
+            )
+        )
+    else:
+        query_job.result()
+        print(
+            "Processed {} bytes to populate {}".format(
+                query_job.total_bytes_processed, destination_table_spec
+            )
+        )
+
+
+def main():
+    args = parser.parse_args()
+
+    client = bigquery.Client()
+    live_datasets = [
+        d
+        for d in client.list_datasets(args.project_id)
+        if d.dataset_id.endswith("_live")
+    ]
+
+    for live_dataset in live_datasets:
+        stable_dataset_id = live_dataset.dataset_id[:-5] + "_stable"
+        for live_table in client.list_tables(live_dataset.reference):
+            live_table_spec = f"{live_table.dataset_id}.{live_table.table_id}"
+            stable_table = client.get_table(
+                ".".join([args.project_id, stable_dataset_id, live_table.table_id])
+            )
+            if args.except_tables is not None and any(
+                fnmatch(live_table_spec, pattern) for pattern in args.except_tables
+            ):
+                print(f"Skipping {live_table_spec} due to --except argument")
+                continue
+            if args.only_tables is not None and not any(
+                fnmatch(live_table_spec, pattern) for pattern in args.only_tables
+            ):
+                print(f"Skipping {live_table_spec} due to --only argument")
+                continue
+            run_deduplication_query(
```

Comment:
feature request: run this using a worker pool, so that getting one slow table won't block all the others

Reply:
Added --parallelism option that gets passed to ThreadPool and defaults to 4.